library/ssl_tls.c

```diff
@@ -1846,6 +1852,92 @@ mbedtls_ssl_mode_t mbedtls_ssl_get_mode_from_ciphersuite(
 }
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_SSL_PROTO_TLS1_3)
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3)
+/* Serialization of TLS 1.3 sessions:
+ *
+ * struct {
+ *   uint64 start_time;
+ *   uint32 ticket_lifetime;
+ *   uint8 ciphersuite[2];
+ *   uint32 ticket_age_add;
+ *   uint8 ticket_flags;
+ *   opaque resumption_key<0..255>;
+ *   opaque ticket<0..2^16>;
+ *   uint64 ticket_received;
+ * } serialized_session_tls13;
+ *
+ */
+#if defined(MBEDTLS_SSL_SESSION_TICKETS)
+
+static size_t ssl_session_save_tls13( const mbedtls_ssl_session *session,
```

Comment:
There is no test for this I think, at least no unit test. It would be great to have some unit testing like we have in TLS 1.2 based on `ssl_serialize_session_save_load` in `test_suite_ssl.function`. We need the load part to enable this though. It could be in this PR or in a separate one, as you want.

Reply:
I'd like  fix it in another PR. That test depends on session load, will add it in separate one.

Reply:
That's ok with me to add the session load together with unit tests in another PR. We may have to revisit the code of `ssl_session_save_tls13` at that time.

Reply:
#6087 is ready for early review